### PR TITLE
feat: test invalid boltz data

### PR DESCRIPTION
**Pull request by jackstar12**
- feat: mock invalid boltz transaction in tests
- feat: transaction confirmation checks
- feat: improve errors

### CODE REVIEW

File: .mockery.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+with-expecter: True
+filename: "{{ .InterfaceName }}_mock.go"
+packages:
+  github.com/BoltzExchange/boltz-client/onchain:
+    interfaces:
+      # select the interfaces you want mocked
```
Comment (michael1011) [inline .mockery.yaml:6]: Do you want to keep those comments?

File: cmd/boltzd/boltzd.go [hunk 2]
```diff
@@ -51,41 +53,60 @@ func main() {
...
+	} else if strings.EqualFold(cfg.Node, "LND") {
+		node = cfg.LND
+	} else if isClnConfigured && isLndConfigured {
+		logger.Fatal("Both CLN and LND are configured. Set --node to specify which node to use.")
+	} else if isClnConfigured {
+		node = cfg.Cln
+	} else if isLndConfigured {
+		node = cfg.LND
+	} else {
+		logger.Fatal("No lightning node configured. Set either CLN or LND.")
```
Comment (michael1011) [inline cmd/boltzd/boltzd.go:79]:
```suggestion
		logger.Fatal("No lightning node configured. Configure either CLN or LND.")
```

File: cmd/boltzd/boltzd.go [hunk 3]
```diff
@@ -51,41 +53,60 @@ func main() {
 	Start(cfg)
 }
 
-func Init(cfg *config.Config) {
+func initLightning(cfg *config.Config) (lightning.LightningNode, error) {
+	if cfg.Standalone {
+		if cfg.Network == "" {
+			logger.Fatal("Standalone mode requires a lightning node to be set")
```
Comment (michael1011) [inline cmd/boltzd/boltzd.go:59]: Shouldn't that say network?

File: mocks/github.com/BoltzExchange/boltz-client/onchain/Wallet_mock.go [hunk 4]
```diff
@@ -0,0 +1,295 @@
+// Code generated by mockery v2.42.2. DO NOT EDIT.
```
Comment (michael1011) [inline mocks/github.com/BoltzExchange/boltz-client/onchain/Wallet_mock.go:1]: Can you add a script for generating those mocks to the Makefile?

File: cmd/boltzd/boltzd_test.go [hunk 5]
```diff
@@ -1680,3 +1827,200 @@ func TestChangePassword(t *testing.T) {
...
+func Test_waitForLightningSynced(t *testing.T) {
+	type args struct {
+		lightning lightning.LightningNode
+	}
+	tests := []struct {
+		name string
+		args args
+		want *lightning.LightningInfo
+	}{
+		// TODO: Add test cases.
```
Comment (michael1011) [inline cmd/boltzd/boltzd_test.go:2019]: There are bunch of these comments in this file

Comment (jackstar12) [reply cmd/boltzd/boltzd_test.go:2019]: accidently had goland generate those

**Review by michael1011 — APPROVED** [review]

**Pull request merged**